src/test/java/com/palantir/docker/compose/TestContainerNames.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.docker.compose;
+
+import static java.util.stream.Collectors.toList;
+
+import com.palantir.docker.compose.connection.ContainerName;
+import com.palantir.docker.compose.connection.ImmutableContainerName;
+import java.util.Arrays;
+import java.util.List;
+
+public class TestContainerNames {
```

Comment:
I think this still has merit, because this one method is used in a bunch of different places to create hacky ContainerName. `ContainerNamesShould` actually creates fully legit ones